src/adapters/store/postgres.ts

```diff
@@ -63,11 +63,28 @@ export function createPgStoreAdapter(pool: Pool, logger: Logger = console): Stor
     }
   }
 
-  async function write(event: Event<EventData, EventContext<any>>) {
+  /**
+   *  This will return an either Left(DuplicateError), Left(PgError) or Right(void);
+   *
+   *
+   */
+  async function write(event: Event<EventData, EventContext<any>>): Promise<Either<DuplicateError | Error, void>> {
     return pool.query(
       'INSERT INTO events(id, data, context) values ($1, $2, $3)',
       [event.id, event.data, event.context],
-    ).then(() => {/**/});
+    )
+    .then(() => Right(undefined))
+    .catch((err) => {
+      if (err instanceof Error) {
+        // duplicate key value violates unique constraint "events_pkey"
+        if (err.message.includes('unique')) {
```

Comment:
It's clearer and more robust to include more of the message in the match I think:

```suggestion
        if (err.message.includes('violates unique constraint')) {
```